src/benchmarks/micro/libraries/System.Text.RegularExpressions/Perf.Regex.Industry.cs

```diff
@@ -0,0 +1,300 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    internal static class Perf_Regex_Industry
+    {
+        public static string ReadInputFile(string name)
+        {
+            string path = Path.Combine(Path.GetDirectoryName(typeof(Perf_Regex_Industry).Assembly.Location), "libraries", "System.Text.RegularExpressions", "TestData", name);
+            using (FileStream fs = File.OpenRead(path))
+            using (var gz = new GZipStream(fs, CompressionMode.Decompress))
+            using (var reader = new StreamReader(gz))
+            {
+                return reader.ReadToEnd();
+            }
+        }
+    }
+
+    /// <summary>Performance tests adapted from https://github.com/mariomka/regex-benchmark</summary>
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Perf_Regex_Industry_Mariomka
+    {
+        [Params(
+            @"[\w\.+-]+@[\w\.-]+\.[\w\.-]+",
+            @"[\w]+://[^/\s?#]+[^\s?#]+(?:\?[^\s#]*)?(?:#[^\s]*)?",
+            @"(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])"
+        )]
+        public string Pattern { get; set; }
+
+        [Params(RegexOptions.None, RegexOptions.Compiled
+#if NET7_0_OR_GREATER
+            , RegexOptions.NonBacktracking
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private string _input_text;
+        private Regex _regex;
+
+        [Benchmark]
+        public Regex Ctor() => new Regex(Pattern, Options);
+
+        [GlobalSetup(Target = nameof(CountMatches))]
+        public void Setup()
+        {
+            _regex = new Regex(Pattern, Options);
+            _input_text = Perf_Regex_Industry.ReadInputFile("mariomka.txt.gz");
+        }
+
+        [Benchmark]
+        public int CountMatches() => _regex.Matches(_input_text).Count;
+    }
+
+    /// <summary>Performance tests adapted from https://github.com/cloudflare/sliceslice-rs/tree/a27b76c8527d44d5b3534c84b878d8289eacb7ff/data</summary>
+    [BenchmarkCategory(Categories.Libraries)]
+    [SkipTooManyTestCasesValidator]
+    public class Perf_Regex_Industry_SliceSlice
+    {
+        [Params(RegexOptions.None, RegexOptions.Compiled
+#if NET7_0_OR_GREATER
+            , RegexOptions.NonBacktracking
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private string _i386;
+        private Regex[] _regexes;
+
+        [GlobalSetup(Target = nameof(CountMatches))]
+        public void Setup()
+        {
+            _i386 = Perf_Regex_Industry.ReadInputFile("i386.txt.gz");
```

Comment:
did you find the license for i386.txt? It says "(c)INTEL CORPORATION 1987" in it but I see no license.

Reply:
unless you think the dynamics are critical I suggest dropping i386.txt.gz and using sherlock.txt.gz here instead.

Reply:
> unless you think the dynamics are critical I suggest dropping i386.txt.gz and using sherlock.txt.gz here instead.

Presumably the words file was derived from this as well.  I expect changing one but not the other would impact the characteristics.

> did you find the license for i386.txt?

I didn't look.  These files came from the repo I linked to and included the license for in the 3rd party licenses file.

Reply:
I removed the i386 and words files.